File: Python/Sort_1.py
def calculate_distance(point_str):
    _, x_str, y_str = point_str.split()
    x, y = int(x_str), int(y_str)
    return x**2 + y**2

t = int(input())
for _ in range(t):
    n = int(input())
    points = []
    for _ in range(n):
        point_str = input()
        points.append(point_str)
    points.sort(key=calculate_distance)
    for point in points:
        print(point)
    input() 

File: Java/Sort_1.java
import java.util.Scanner;
import java.util.Arrays;
import java.util.Comparator;

class Main 
{
    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);
        int t = input.nextInt();
        input.nextLine();
        for (int i = 0; i < t; i++) 
        {
            int n = input.nextInt();
            input.nextLine();
            String[][] points = new String[n][3];
            for (int j = 0; j < n; j++) 
            {
                String pointStr = input.nextLine();
                String[] data = pointStr.split(" ");
                int x = Integer.parseInt(data[1]);
                int y = Integer.parseInt(data[2]);
                int distance = x * x + y * y;
                points[j] = new String[]{pointStr, String.valueOf(distance)};
            }
            Arrays.sort(points, Comparator.comparingInt(p -> Integer.parseInt(p[1])));
            for (String[] point : points)
                System.out.println(point[0]);
        }
        input.close();
    }
}
